src/main/perl/Lock.pm

```diff
@@ -193,6 +235,61 @@ sub _try_lock
     return SUCCESS;
 }
 
+
+# Before CAF PR#132, CAF::Lock used the presence of the lock file as
+# condition to determine if the lock was taken or not.
+# The file would hold the PID of the process that had the lock
+# (and unlock simply removed the file).
+# This method provides support for the existence of an old-style lockfile
+# held by another process using old-style locking code.
+# It does not modify/cleanup the old-style file (even if it is stale).
+# Returns boolean true if the lock has been taken by another process
+# (i.e. if it is locked by another old-style instance).
+sub _is_locked_oldstyle
+{
+    my ($self, $force) = @_;
+
+    my $other_has_lock = 0;
+
+    # No file -> no lock
+    # Empty file (like a lock with current code) -> no old-style lock
+    my @st = stat($self->{LOCK_FILE});
```

Comment:
Why use stat here rather than just -e ? The long if() statement requires a lot of reading to understand nothing happens. I think it would be better to "short circuit" this sub in the routine case:
```
# use stat to detect file existence because ...
my @st = stat($self->{LOCK_FILE});
if( ! @st && $st[7] > 0 ) {
  # No file -> no lock
  return 0;
} else {
  my $other_has_lock = 0;
  <<current code>>
}
```